spec/controllers/collection_items_controller_spec.rb

```diff
@@ -119,4 +119,211 @@
+        before { fake_login }
+
+        it "errors and redirects" do
+          patch :update_multiple, params: params
+          it_redirects_to_with_error(collection, "You don't have permission to do that, sorry!")
+        end
+      end
+
+      context "when logged in as a maintainer" do
+        before { fake_login_known_user(collection.owners.first.user) }
+
+        {
+          collection_approval_status: -1,
+          unrevealed: true,
+          anonymous: true
+        }.each_pair do |field, value|
+          context "setting #{field}" do
+            let(:attributes) { { field => value } }
+
+            it "updates the collection item and redirects" do
+              patch :update_multiple, params: params
+              expect(item.reload.send(field)).to eq(value)
+              it_redirects_to_with_notice(collection_items_path(collection),
+                                          "Collection status updated!")
+            end
+          end
+        end
+
+        context "setting remove" do
+          let(:attributes) { { remove: "1" } }
+
+          it "deletes the collection item and redirects" do
+            patch :update_multiple, params: params
+            expect { item.reload }.to \
+              raise_exception(ActiveRecord::RecordNotFound)
+            it_redirects_to_with_notice(collection_items_path(collection),
+                                        "Collection status updated!")
+          end
+        end
+
+        context "setting user_approval_status" do
+          let(:attributes) { { user_approval_status: -1 } }
+
+          it "throws an error and doesn't update" do
+            expect do
+              patch :update_multiple, params: params
+            end.to raise_exception(ActionController::UnpermittedParameters)
+            expect(item.reload.user_approval_status).not_to eq(-1)
+          end
+        end
+      end
+    end
+
+    describe "on the collection items page for a different user" do
+      let(:user) { create(:user) }
+      before { fake_login_known_user(user) }
+
+      let(:params) do
+        {
+          user_id: user.login,
+          collection_items: {
+            item.id => { user_approval_status: -1 }
+          }
+        }
+      end
+
+      it "silently fails to update the collection item" do
+        patch :update_multiple, params: params
+        expect(item.reload.user_approval_status).not_to eq(-1)
+        it_redirects_to_with_notice(user_collection_items_path(user),
+                                    "Collection status updated!")
+      end
+    end
+
+    describe "on the collection items page for a different collection" do
+      let(:other_collection) { create(:collection) }
+      before { fake_login_known_user(other_collection.owners.first.user) }
+
+      let(:params) do
+        {
+          collection_id: other_collection.name,
+          collection_items: {
+            item.id => { collection_approval_status: -1 }
```

Comment:
Should we use `CollectionItem::REJECTED` instead of -1?

Reply:
Good idea, thanks! Will do.